R/estimateDiversity.R

```diff
@@ -240,229 +248,273 @@ NULL
-        }
-
-        # input check
-        supported_index <- c("coverage", "fisher", "gini_simpson", 
-                             "inverse_simpson", "log_modulo_skewness", "shannon")
-        index_string <- paste0("'", paste0(supported_index, collapse = "', '"), "'")
-        if ( !all(index %in% supported_index) || !(length(index) > 0)) {
-            stop("'", paste0("'index' must be from the following options: '",index_string), "'", call. = FALSE)
-        }
-        
-        if(!.is_non_empty_character(name) || length(name) != length(index)){
-            stop("'name' must be a non-empty character value and have the ",
-                "same length as 'index'.",
-                call. = FALSE)
-        }
-        .check_assay_present(assay.type, x)
-        .require_package("vegan")
-
-        dvrsts <- BiocParallel::bplapply(index,
-                                        .get_diversity_values,
-                                        x = x,
-                                        mat = assay(x, assay.type),
-                                        BPPARAM = BPPARAM,
-                                        ...)
-        .add_values_to_colData(x, dvrsts, name)
-    }
+          function(x, assay.type = "counts", assay_name = NULL,
+                   index = c("coverage", "fisher", "gini_simpson", 
+                             "inverse_simpson", "log_modulo_skewness", "shannon"),
+                   name = index, ..., BPPARAM = SerialParam()){
+              
+              if (!is.null(assay_name)) {
+                  .Deprecated(old="assay_name", new="assay.type", 
+                              "Now assay_name is deprecated. Use assay.type instead.")
+              }
+              
+              # input check
+              supported_index <- c("coverage", "fisher", "gini_simpson", 
+                                   "inverse_simpson", "log_modulo_skewness", "shannon")
+              index_string <- paste0("'", paste0(supported_index, collapse = "', '"), "'")
+              if ( !all(index %in% supported_index) || !(length(index) > 0)) {
+                  stop("'", paste0("'index' must be from the following options: '",index_string), "'", call. = FALSE)
+              }
+              
+              if(!.is_non_empty_character(name) || length(name) != length(index)){
+                  stop("'name' must be a non-empty character value and have the ",
+                       "same length as 'index'.",
+                       call. = FALSE)
+              }
+              .check_assay_present(assay.type, x)
+              .require_package("vegan")
+              
+              dvrsts <- BiocParallel::bplapply(index,
+                                               .get_diversity_values,
+                                               x = x,
+                                               mat = assay(x, assay.type),
+                                               BPPARAM = BPPARAM,
```

Comment:
Only the indentation is changed. Also it seems that the indentation is not multipication of 4 spaces anymore

1. The indentation should be multiplication of 4
2. We do not want to change code like this. It looks like the whole code has changed. As people are already using the package, we must retain the code. It is extremely hard to see the history of code when there are this kind of changes that do not really affect the code. --> indentation is fixed only when the code is also changed

Reply:
Okay fixed !